lib/Doctrine/DBAL/Statement.php

```diff
@@ -138,6 +138,8 @@ public function bindValue($name, $value, $type = null)
      */
     public function bindParam($name, &$var, $type = PDO::PARAM_STR, $length = null)
     {
+        $this->params[$name] = $var;
+        $this->types[$name] = $type;
         return $this->stmt->bindParam($name, $var, $type, $length);
```

Comment:
Can you please add a newline before the `return` statement? Otherwise patch looks good to me.